### PR TITLE
- added parsers for btrfs tools

### PR DESCRIPTION
**Pull request by aschnell**
For https://trello.com/c/yES1u2VG/425-8-libstorageng-btrfs-subvolumes.

### CODE REVIEW

File: storage/SystemInfo/CmdBtrfs.cc [hunk 1]
```diff
@@ -194,19 +192,59 @@ namespace storage
...
-	    s << "data " << entry.path << '\n';
+	s << "id:" << entry.id << " parent-id:" << entry.parent_id
+	  << " path:" << entry.path << '\n';
 
 	return s;
     }
 
 
+    CmdBtrfsSubvolumeGetDefault::CmdBtrfsSubvolumeGetDefault(const string& mount_point)
+	: id(-1 /* BtrfsSubvolume::Impl::unknown_id */)
```
Comment (joseivanlopez) [inline storage/SystemInfo/CmdBtrfs.cc]: Does `BtrfsSubvolume::Impl::unknown_id` exist? or is it only a reference for future implementation?

File: storage/Utils/AppUtil.h [hunk 2]
```diff
@@ -96,6 +96,7 @@ void classic(StreamType& stream)
     string hostname();
     string datetime( time_t t1, bool utc=false, bool classic=true );
 
+    string generated_string();
```
Comment (aschnell) [inline storage/Utils/AppUtil.h]: It is a global function. Those cannot have a const attribute.

File: examples/SystemInfo/test-cmd-btrfs-filesystem-show.cc [hunk 3]
```diff
@@ -0,0 +1,34 @@
...
+using namespace std;
+using namespace storage;
+
+
+void
+test_cmd_btrfs_filesystem_show(SystemInfo& systeminfo)
+{
+    try
+    {
+	const CmdBtrfsFilesystemShow& cmdbtrfsfilesystemshow = systeminfo.getCmdBtrfsFilesystemShow();
```
Comment (aschnell) [inline examples/SystemInfo/test-cmd-btrfs-filesystem-show.cc]: Yes, partly I am renaming those. But this has very low priority for me.

File: storage/SystemInfo/CmdBtrfs.cc (hunk 1 above)
Comment (aschnell) [inline storage/SystemInfo/CmdBtrfs.cc]: Indeed, ```BtrfsSubvolume::Impl::unknown_id``` will be added later.

**Pull request merged**